scapy/layers/inet6.py

```diff
@@ -498,7 +507,7 @@ def answers(self, other):
         if not isinstance(other, IPv6): # self is reply, other is request
             return False
         if conf.checkIPaddr:
-            ss = inet_pton(socket.AF_INET6, self.src)
+            # ss = inet_pton(socket.AF_INET6, self.src)
```

Comment:
I have let that as a comment so that we understand the comment in `elif (sd != os): # or ss != od): <- removed for ICMP errors` some lines after.

Reply:
I can remove both comments if you prefer, but I think it's ok to leave them (to prevent someone from reintroducing this test later). Let me know what you think.